### PR TITLE
chore: support WS2022 and WS2022-gen2 in AgentBaker Windows E2E

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind improvement

**What this PR does / why we need it**:
Support WS2022 and WS2022-gen2 in AgentBaker Windows E2E

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: .pipelines/e2e-windows-template.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+parameters:
```
Comment (AbelHu) [inline .pipelines/e2e-windows-template.yaml:1]: Can we move this file to `.pipelines/templates/` since it is a template?

Comment (junjiezhang1997) [reply .pipelines/e2e-windows-template.yaml:1]: Sure. That's better.

File: .pipelines/e2e-windows-template.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  - name: artifactName
```
Comment (AbelHu) [inline .pipelines/e2e-windows-template.yaml:2]: windowsDistro or windowsImage?

File: e2e/e2e-windows-scenario.sh [hunk 3]
```diff
@@ -132,22 +134,28 @@ MC_WIN_VMSS_NAME=$(az vmss list -g $MC_RESOURCE_GROUP_NAME --query "[?contains(n
 VMSS_RESOURCE_Id=$(az resource show --resource-group $MC_RESOURCE_GROUP_NAME --name $MC_WIN_VMSS_NAME --resource-type Microsoft.Compute/virtualMachineScaleSets --query id --output tsv)
 
 az group export --resource-group $MC_RESOURCE_GROUP_NAME --resource-ids $VMSS_RESOURCE_Id --include-parameter-default-value > test.json
+# TODO: ues prod image
```
Comment (AbelHu) [inline e2e/e2e-windows-scenario.sh:137]: typo: use

File: .pipelines/e2e-windows-template.yaml (hunk 2 above)
Comment (junjiezhang1997) [reply .pipelines/e2e-windows-template.yaml]: Modified

File: e2e/e2e-windows-scenario.sh (hunk 3 above)
Comment (junjiezhang1997) [reply e2e/e2e-windows-scenario.sh]: Modified

File: e2e/windows_vmss_template.json [hunk 4]
```diff
@@ -6,9 +6,9 @@
             "defaultValue": "",
             "type": "string"
         },
-        "galleries_windowsabgallery_externalid": {
-            "defaultValue": "/subscriptions/a15c116e-99e3-4c59-aebc-8f864929b4a0/resourceGroups/akswinvhdbuilderrg/providers/Microsoft.Compute/galleries/AKSWindows",
-            "type": "string"
+        "galleries_AKSWindows_externalid": {
```
Comment (AbelHu) [inline e2e/windows_vmss_template.json:9]: do we still need this parameter?

Comment (junjiezhang1997) [reply e2e/windows_vmss_template.json:9]: yes, I set the value of it in e2e-windows-scenario.sh

File: e2e/e2e-create-windows-nodepool.sh [hunk 5]
```diff
@@ -6,13 +6,24 @@ source e2e-helper.sh
...
-if [ "$out" == null ]; then
+if [[ "$retval" != "0" ]]; then
     log "Creating windows nodepool"
-    # If you want to specify the k8s version for windows nodepool, you should also specify the version when creating cluster
-    # and modify the value of "customKubeBinaryURL" and "customKubeProxyImage" in property-windows.json
-    az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name $WINDOWS_NODEPOOL --os-type Windows --node-count 1
+    if [ "$WINDOWS_E2E_IMAGE" == "2019-containerd" ]; then
+        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --node-count 1
+    elif [ "$WINDOWS_E2E_IMAGE" == "2022-containerd" ]; then
+        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku Windows2022 --node-count 1
```
Comment (AbelHu) [inline e2e/e2e-create-windows-nodepool.sh:21]: Does the default VMSize only support gen1?

Comment (junjiezhang1997) [reply e2e/e2e-create-windows-nodepool.sh:21]: No, default VMSize is Standard_D2s_v3. It supports both gen1 and gen2. Should I use a VMSize that only supports gen1 as default VMSize?

Comment (AbelHu) [reply e2e/e2e-create-windows-nodepool.sh:21]: yes. otherwise ws2022 gen1 will not be covered.

File: e2e/e2e-create-windows-nodepool.sh [hunk 6]
```diff
@@ -6,13 +6,23 @@ source e2e-helper.sh
...
-    # and modify the value of "customKubeBinaryURL" and "customKubeProxyImage" in property-windows.json
-    az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name $WINDOWS_NODEPOOL --os-type Windows --node-count 1
+    if [ "$WINDOWS_E2E_IMAGE" == "2019-containerd" ]; then
+        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --node-count 1
+    elif [ "$WINDOWS_E2E_IMAGE" == "2022-containerd" ]; then
+        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku Windows2022 --node-vm-size Standard_D2_v2 --node-count 1
+    elif [ "$WINDOWS_E2E_IMAGE" == "2022-containerd-gen2" ]; then
+        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku Windows2022 --node-vm-size Standard_DS2_v2 --node-count 1
+    else
+        err "Invalid windows e2e image version"
```
Comment (AbelHu) [inline e2e/e2e-create-windows-nodepool.sh:24]: Will it exit after log?

Comment (junjiezhang1997) [reply e2e/e2e-create-windows-nodepool.sh:24]: Add exit code for invalid case.

**Review by AbelHu — APPROVED** [review]

**Pull request merged**